src/core/Akka/Serialization/Serializer.cs

```diff
@@ -137,7 +137,7 @@ protected SerializerWithStringManifest(ExtendedActorSystem system) : base(system
         /// <returns>The object contained in the array</returns>
         public sealed override object FromBinary(byte[] bytes, Type type)
         {
-            var manifest = type.TypeQualifiedName();
```

Comment:
There's a bug here, `SerializerWithStringManifest` uses different manifest in this code path

Reply:
This method should not be called on `SerializerWithStringManifest` unless an end-user directly casts it into a `Serializer`, so I don't think this needs to be changed for this test.